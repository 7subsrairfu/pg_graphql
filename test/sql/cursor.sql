begin;
    -- Check SQL output
    select graphql.to_cursor_clause(
        'abc',
        array[('email', 'asc', true), ('id', 'asc', false)]::graphql.column_order_w_type[]
    );


    -- Check encode/decode round trip
    with c(ursor) as (
        select
            '("{""(email,asc,t)"",""(id,asc,f)""}","[""aardvark@x.com"", 1]")'::graphql.cursor
    )
    select
        c.ursor =
        graphql.decode(
            graphql.encode(
                c.ursor::graphql.cursor
            )
        )
    from c;

rollback
