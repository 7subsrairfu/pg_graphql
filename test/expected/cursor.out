begin;
	-- Check SQL output
	select graphql.to_cursor_clause(
		'abc',
		array[('email', 'asc', true), ('id', 'asc', false)]::graphql.column_order[]
	);
                                     to_cursor_clause                                      
-------------------------------------------------------------------------------------------
 (                                                                                        +
                                 '{"(email,asc,t)","(id,asc,f)"}'::graphql.column_order[],+
                                 jsonb_build_array(to_jsonb(abc.email), to_jsonb(abc.id)) +
                         )::graphql.cursor
(1 row)

	
	-- Check encode/decode round trip	
    with c(ursor) as (
        select 
            '("{""(email,asc,t)"",""(id,asc,f)""}","[""aardvark@x.com"", 1]")'::graphql.cursor
    )
    select 	
        c.ursor =
        graphql.decode(
            graphql.encode(
                c.ursor::graphql.cursor
            )
        )
    from c;
 ?column? 
----------
 t
(1 row)

rollback
